### PR TITLE
Dockerize project

### PR DESCRIPTION
**Pull request by dresswithpockets**
Local environment building and testing using docker.
Runs postgres and bot as two separate processes under the same container.

### CODE REVIEW

File: src/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# Setup a Development User
+RUN useradd -ms /bin/bash sbubby 
+USER sbubby
+
+# copy project
+COPY . .
+
+# Run Commands
+ENTRYPOINT ["/opt/api/cmds/entrypoint.sh"]
```
Comment (PseudonymPatel) [inline src/Dockerfile:30]: I had to make this `ENTRYPOINT ["bash", "opt/api/cmds/entrypoint.sh"]` for it to work.

File: src/cmds/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+cd /opt/api
+
+# The entrypoint for the Postscript Homework API
+./cmds/wait-for-it.sh db:5432 -s -- printf "Database Successfully Started\n"
```
Comment (PseudonymPatel) [inline src/cmds/entrypoint.sh:6]: I get a "permission denied", although this is probably just a problem with my setup
```
> docker run sbubby
/opt/api/cmds/entrypoint.sh: line 6: ./cmds/wait-for-it.sh: permission denied
```

**Review by PseudonymPatel — CHANGES_REQUESTED** [review]

File: src/cmds/entrypoint.sh (hunk 2 above)
Comment (dresswithpockets) [reply src/cmds/entrypoint.sh:6]: Did you install docker as admin?

File: src/Dockerfile (hunk 1 above)
Comment (dresswithpockets) [reply src/Dockerfile:30]: what OS are you on?

Comment (PseudonymPatel) [reply src/Dockerfile:30]: windows

Comment (PseudonymPatel) [reply src/Dockerfile:30]: Ubuntu 20.04, in a vm

File: src/cmds/entrypoint.sh (hunk 2 above)
Comment (PseudonymPatel) [reply src/cmds/entrypoint.sh:6]: I installed it through the command line with sudo

File: src/Dockerfile (hunk 1 above)
Comment (dresswithpockets) [reply src/Dockerfile:30]: Hmm, you defo shouldnt need to exec BASH directly here. I think the problem is that `entrypoint.sh` doesn't have exec perms by default - so, ill look at docker docs to see if theres a way to apply that perm directly, or if we need to make the change you're suggesting.

File: src/cmds/entrypoint.sh (hunk 2 above)
Comment (dresswithpockets) [reply src/cmds/entrypoint.sh:6]: wait, so, entrypoint.sh actually executes, you just need to have the ENTRYPOINT be bash in order to not get this perm error?

File: src/Dockerfile (hunk 1 above)
Comment (PseudonymPatel) [reply src/Dockerfile:30]: had to use docker-compose

**Review by PseudonymPatel — APPROVED** [review]

**Comment by PseudonymPatel** [comment]
Didn't use docker-compose, everything runs smoothly

**Pull request merged**